ssl/tls_srp.c

```diff
@@ -132,11 +134,17 @@ int SSL_SRP_CTX_init(struct ssl_st *s)
         SSLerr(SSL_F_SSL_SRP_CTX_INIT, ERR_R_INTERNAL_ERROR);
         goto err;
     }
+    if ((ctx->srp_ctx.info != NULL) &&
+        ((s->srp_ctx.info = BUF_strdup(ctx->srp_ctx.info)) == NULL)) {
+        SSLerr(SSL_F_SSL_SRP_CTX_INIT, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
     s->srp_ctx.srp_Mask = ctx->srp_ctx.srp_Mask;
 
     return (1);
  err:
     OPENSSL_free(s->srp_ctx.login);
+    OPENSSL_free(s->srp_ctx.info);
```

Comment:
Using memset could be safer yes, but I did the least invasive fix so that it was clear what was being changed.

I can do a second commit changing all those NULL assignments into a memset if you prefer, or is there another OpenSSL specific call that should be used instead of memset()? I am not familiar with OpenSSL internals.